### PR TITLE
[SEMI-MODULAR] MCR Expansion Vol 1 & Balance Passes

### PR DESCRIPTION
**Pull request by MassacreDemigirl**
## About The Pull Request
Update: Made inline with the refactors of #11536

This PR started with realizing MCR's did 27 full laser damage and evolved into what it is right now:

Firstly it separates the MCR Laser damage to 20.
Nerfs the laser damage from 27 to 22
Makes the undercharger no longer give grip bonuses when activated.
Repeater beams do 15 damage each for a total of 30 damage.
Scatter beams do 10 damage each for a total of 30 damage.
Moves the superheater upgrade to an illegal MCR attachments tree.
Superheater lasers do 15 damage but still keep 2 firestacks.
Adds Hellfire upgrade in place of superheater, deals 25 damage and has the wound bonuses of hellfire lasers.
Adds a Lance upgrade that gives the MCR the firing delay of 2 seconds however it fires a Heavy Laser beam at twice the projectile speed.
Adds an Illegal Repeater upgrade that does the same damage but has 50 armor penetration.
Adds an Illegal Scatter upgrade that shoots 7 pellets for a total of 45 damage.
Adds an Illegal suppressor upgrade. It suppresses the laser somehow.
Adds a Clown Tech extension for MCR's, a bikehorn upgrade that makes you shoot yellow beams of slipping and low stamina damage.
Adds 3 new skins to the MCR
Lowers the attachment slots of Advanced Cells to 2 and BS Cells to 3
Makes loading a new Cell into the MCR take 4 seconds while tactical reloading takes 6 seconds.
Adds a reloading assister attachment that makes loading a cell take 2 seconds.
Adds a tac-reload helping attachment that makes tactical reloads 3 seconds.

## How This Contributes To The Skyrat Roleplay Experience
When MCR's were merged it caused an uprise in usage of lasers. However something was odd, they were somehow far too strong. Yet there was no code editing the damage of lasers in the MCR PR. This is apparently because of the upgraded base Beam Projectile damage. Which was not edited in any of its childs. Thus more used laser weapons were not affected by this buff (Captains Lasgun and currently Blueshields Lasgun), as well as the Lasgun and Egun were not preferred weapons due to Ballistics and further on Hardlight Bullets existing. Thus this effect was not felt till the merging of MCRs. 

Here are a few other projectiles to compare to:
```
/obj/projectile/bullet/c32
    name = ".32 bullet"
    damage = 15
/obj/projectile/bullet/c46x30mm
    name = "4.6x30mm bullet"
    damage = 20
/obj/projectile/beam/laser/hellfire
	name = "hellfire laser"
	wound_bonus = 0
	damage = 25
/obj/projectile/beam/laser
	damage = 27 // SKYRAT EDIT: 20
	wound_bonus = -30
	bare_wound_bonus = 40
```
As you can see rest of the weaponry were never changed and were in a good spot. However lasers being buffed the same amount of the HP upgrade was, meant that they did not follow the lethality rating of the rest of the guns.

Before this PR, a laser type weapon takes 5 hits to crit an unarmored person with a damage of 27.
Whereas normal bullets and the lasers in this merge takes 6.75 hits to crit someone with a damage of 20.

This PR should bring MCR's to a closer line in Skyrat Combat as the damage upgrade was the equivalent of upping the damage of 357 to 81 or in a way making the HP pool of people 100 just for that gun in its balance.

Normal lasers are not touched to keep them viable and also to not make HoS's egun utter waste.

___
MCR's suffered from a lack of meta availability of mods. The only thing you would see be used were superheaters or repeaters with grips. This PR hopes to make MCR modding an actual choice instead of just being put all available mods on the cells and use one of the two good choices for the MCR. 

By adding the same type of loading time tactical reloads have on to normal reloads should balance the fact that people can carry an entire armory worth of shots in their backpack and pockets. 

By decreasing the mod slots of Advanced and BS cells this gives the players an actual choice to make along with the new reload helpers and the loading time.

Moving the superheater upgrade along with adding a few more buffed versions of the repeater and scatter as well as a few bonus items makes them harder to acquire. 

Increasing the price of the self charging attachment should turn it into an actual investment rather then a cheap addon that instantly makes the gun good.

Overall most heat and power values along with material costs have been tweaked to make MCR's more in line with other available weaponry.

## Changelog

:cl:ErdinyoBarboza, ToryJ for making me code dive
balance: Nerfed Base Laser damage to 22 from 27
balance: Nerfed the MCR Laser damage to 20.
balance: Made the undercharger no longer give grip bonuses when activated.
balance: Repeater beams do 15 damage each for a total of 30 damage.
balance: Scatter beams do 10 damage each for a total of 30 damage.
balance: Moves the superheater upgrade to an illegal MCR attachments tree.
balance: Superheater lasers do 15 damage but still keep 2 firestacks.
expansion: Adds Hellfire upgrade in place of superheater, deals 25 damage and has the wound bonuses of hellfire lasers.
expansion: Adds a Lance upgrade that gives the MCR the firing delay of 2 seconds however it fires a Heavy Laser beam at twice the projectile speed.
expansion: Adds an Illegal Repeater upgrade that does the same damage but has 50 armor penetration.
expansion: Adds an Illegal Scatter upgrade that shoots 7 pellets for a total of 45 damage.
expansion: Adds an Illegal suppressor upgrade. It suppresses the laser somehow.
expansion: Adds two new cargo crates for MCR attachments.
expansion: Adds a Clown Tech extension for MCR's, a bikehorn upgrade that makes you shoot yellow beams of slipping and low stamina damage.
expansion: Adds 3 new skins to the MCR.
balance: Lowers the attachment slots of Advanced Cells to 2 and BS Cells to 3.
balance: Makes loading a new Cell into the MCR take 4 seconds while tactical reloading takes 6 seconds.
balance: Adds a reloading assister attachment that makes loading a cell take 2 seconds.
balance: Adds a tac-reload helping attachment that makes tactical reloads 3 seconds.
fix: Makes the MCR ammo boxes use their actual sprite.
spellcheck: Fixes some of the comments in the MCR Module.
imageadd: Adds sprites to the afformentioned additions.
/:cl:

### CODE REVIEW

**Comment by Zenitheevee** [comment]
Redacted due to more information-
Good shit here so far.

File: modular_skyrat/modules/microfusion/code/microfusion_gun_attachments.dm [hunk 1]
```diff
@@ -83,6 +83,7 @@ The cell is stable and will not emit sparks when firing.
 	var/recoil_to_add = 1
 	/// The spread to add.
 	var/spread_to_add = 10
+	var/projectile_override =/obj/projectile/beam/laser/microfusion/scatter
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/microfusion/code/microfusion_gun_attachments.dm]:
```suggestion
	var/projectile_override = /obj/projectile/beam/laser/microfusion/scatter
```

File: modular_skyrat/modules/microfusion/code/microfusion_gun_attachments.dm [hunk 2]
```diff
@@ -130,6 +132,35 @@ The gun can fire volleys of shots.
...
+/obj/item/microfusion_gun_attachment/hellfire
+	name = "hellfire emitter upgrade"
+	desc = "A barrel attachment hooked to the phase emitter, this adjusts the beam's wavelength to carry an extra wave of heat; causing nastier wounds and more damage."
+	icon_state = "attachment_hellfire"
+	attachment_overlay_icon_state = "attachment_hellfire"
+	incompatable_attachments = list(/obj/item/microfusion_gun_attachment/scatter, /obj/item/microfusion_gun_attachment/superheat)
+	heat_addition = 70
+	power_usage = 20
+	slot = GUN_SLOT_BARREL
+	var/projectile_override =/obj/projectile/beam/laser/microfusion/hellfire
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/microfusion/code/microfusion_gun_attachments.dm:190]:
```suggestion
	var/projectile_override = /obj/projectile/beam/laser/microfusion/hellfire
```

File: modular_skyrat/modules/microfusion/code/microfusion_gun_attachments.dm [hunk 3]
```diff
@@ -103,19 +105,19 @@ The cell is stable and will not emit sparks when firing.
...
 	name = "superheating phase emitter upgrade"
 	desc = "A barrel attachment hooked to the phase emitter, this adjusts the beam's wavelength to carry an intense wave of heat; causing targets to ignite."
 	icon_state = "attachment_superheat"
 	attachment_overlay_icon_state = "attachment_superheat"
-	incompatable_attachments = list(/obj/item/microfusion_gun_attachment/scatter)
+	incompatable_attachments = list(/obj/item/microfusion_gun_attachment/scatter, /obj/item/microfusion_gun_attachment/hellfire)
 	heat_addition = 70
 	slot = GUN_SLOT_BARREL
-	var/projectile_override =/obj/projectile/beam/laser/microfusion/superheated
+	var/projectile_override =/obj/projectile/beam/laser/microfusion/scatter
```
Comment (Gandalf2k15) [inline modular_skyrat/modules/microfusion/code/microfusion_gun_attachments.dm:161]:
```suggestion
	var/projectile_override = /obj/projectile/beam/laser/microfusion/superheated
```

File: modular_skyrat/modules/microfusion/code/microfusion_gun_attachments.dm [hunk 4]
```diff
@@ -130,6 +132,35 @@ The gun can fire volleys of shots.
 	. = ..()
 	chambered.loaded_projectile = new projectile_override
 
+/*
+HELLFIRE ATTACHMENT
+
+The gun can fire volleys of shots.
```
Comment (Zonespace27) [inline modular_skyrat/modules/microfusion/code/microfusion_gun_attachments.dm]: wrong comment i think

File: modular_skyrat/modules/microfusion/code/cargo_stuff.dm [hunk 5]
```diff
@@ -35,17 +35,53 @@
 	crate_name = "Microfusion Cell Crate"
 
 /datum/supply_pack/security/mcr01_attachments
-	name = "MCR-01 Military Attachments Crate"
-	desc = "Micron Control Systems Incorporated supplied MCR-01 Military spec attachments!"
-	cost = CARGO_CRATE_VALUE * 15
+	name = "MCR-01 Military Attachments Crate Type A"
+	desc = "Micron Control Systems Incorporated supplied MCR-01 Military spec attachments! This crate comes in a utilitarian repeater loadout."
```
Comment (ghost) [inline modular_skyrat/modules/microfusion/code/cargo_stuff.dm]:
```suggestion
	desc = "Micron Control Systems Incorporated supplied MCR-01 Military spec attachments! This crate comes with two utilitarian repeater loadout."
```

File: modular_skyrat/modules/microfusion/code/microfusion_designs.dm [hunk 6]
```diff
@@ -226,3 +316,23 @@
...
+	category = list("Weapons")
+	departmental_flags = DEPARTMENTAL_FLAG_SECURITY | DEPARTMENTAL_FLAG_SCIENCE
+
+/datum/design/microfusion_gun_attachment_honk
+	name = "Bananium Phase Emitter Upgrade"
+	desc = "Splits the microfusion laser beam entering the lens even more!"
+	id = "microfusion_gun_attachment_suppressor"
+	build_type = PROTOLATHE | AWAY_LATHE | MECHFAB
+	materials = list(/datum/material/iron = 1000, /datum/material/glass = 1000, /datum/material/bananium = 1000)
+	build_path = /obj/item/microfusion_gun_attachment/suppressor
```
Comment (ghost) [inline modular_skyrat/modules/microfusion/code/microfusion_designs.dm]: Desc, ID, and Build_path arent to the proper things

File: modular_skyrat/modules/microfusion/code/microfusion_designs.dm [hunk 7]
```diff
@@ -226,3 +316,23 @@
 	build_path = /obj/item/microfusion_gun_attachment/rgb
 	category = list("Weapons")
 	departmental_flags = DEPARTMENTAL_FLAG_SECURITY | DEPARTMENTAL_FLAG_SCIENCE
+
+/datum/design/microfusion_gun_attachment_suppressor
+	name = "Crystalline Diffuser Microfusion Lens Attachment"
+	desc = "Splits the microfusion laser beam entering the lens even more!"
+	id = "microfusion_gun_attachment_suppressor"
```
Comment (ghost) [inline modular_skyrat/modules/microfusion/code/microfusion_designs.dm]: description is wrong

File: modular_skyrat/modules/microfusion/code/microfusion_energy_master.dm [hunk 8]
```diff
@@ -233,6 +235,7 @@
 	if (.)
 		return
 	if(istype(attacking_item, base_cell_type))
+		//reload_timer(user, attacking_item)
```
Comment (ghost) [inline modular_skyrat/modules/microfusion/code/microfusion_energy_master.dm]: do we need this commented out code?

File: modular_skyrat/modules/microfusion/code/microfusion_energy_master.dm [hunk 9]
```diff
@@ -624,8 +629,15 @@
...
-		to_chat(user, span_notice("You insert [inserting_cell] into [src]!"))
+	else
+		if(display_message)
+			to_chat(user, span_notice("You start to insert [inserting_cell] into [src]!"))
+		if(!do_after(user, reload_time_slow, src))
+			if(display_message)
+				to_chat(user, span_warning("You fail to insert [inserting_cell] into [src]!"))
+			return FALSE
+		if(display_message)
+			to_chat(user, span_notice("You insert [inserting_cell] into [src]!"))
```
Comment (ghost) [inline modular_skyrat/modules/microfusion/code/microfusion_energy_master.dm:639]: would be easier if this just early-returned for if(!display_message)

File: modular_skyrat/modules/microfusion/code/microfusion_techweb.dm [hunk 10]
```diff
@@ -87,3 +91,36 @@
 		"microfusion_gun_attachment_xray",
 	)
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 15000)
+
+// Warcrime microfusion
+/datum/techweb_node/quantum_microfusion
```
Comment (ghost) [inline modular_skyrat/modules/microfusion/code/microfusion_techweb.dm]: path is outdated

**Review by Gandalf2k15 — CHANGES_REQUESTED** [review]
Fix up requested changes

**Comment by MassacreDemigirl** [comment]
@Gandalf2k15 its ready

**Comment by Gandalf2k15** [comment]
[09:40:33] Runtime in stack_trace.dm,3: Invalid techweb nodes detected
  proc name: stack trace (/proc/stack_trace)
  src: null
  call stack:
  stack trace("Invalid techweb nodes detected")
  Research (/datum/controller/subsystem/research): initialize all techweb nodes(0)
  Research (/datum/controller/subsystem/research): Initialize(3[48](https://github.com/Skyrat-SS13/Skyrat-tg/runs/5213960866?check_suite_focus=true#step:7:48)329)
  Master (/datum/controller/master): Initialize(10, 0, 1)

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**